cordova-lib/src/cordova/run.js

```diff
@@ -32,8 +32,10 @@ module.exports = function run(options) {
         var hooksRunner = new HooksRunner(projectRoot);
         return hooksRunner.fire('before_run', options)
         .then(function() {
-            // Run a prepare first, then shell out to run
-            return require('./cordova').raw.prepare(options);
+            if (!options.options.noprepare) {
+                // Run a prepare first, then shell out to run
+                return require('./cordova').raw.prepare(options);
+            }

```

Comment:
Are you sure that's necessary? Any promise libraries I've used did not require returning another promise when the then function completes synchronously. Even for a similar case in this file, that is not done after the if block on line 72.